src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -809,6 +1076,1008 @@ static void Test(out System.Collections.Generic.IEnumerable<System.Int32> x)
+                //         Test2(x[out var x1], x1);
+                Diagnostic(ErrorCode.ERR_SyntaxError, "x1").WithArguments(",", "").WithLocation(7, 25),
+                // (7,21): error CS0103: The name 'var' does not exist in the current context
+                //         Test2(x[out var x1], x1);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "var").WithArguments("var").WithLocation(7, 21),
+                // (7,25): error CS0103: The name 'x1' does not exist in the current context
+                //         Test2(x[out var x1], x1);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "x1").WithArguments("x1").WithLocation(7, 25),
+                // (7,30): error CS0103: The name 'x1' does not exist in the current context
+                //         Test2(x[out var x1], x1);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "x1").WithArguments("x1").WithLocation(7, 30)
+                );
+
+            var tree = compilation.SyntaxTrees.Single();
+            Assert.False(tree.GetRoot().DescendantNodes().OfType<ArgumentSyntax>().Where(p => p.Identifier.ValueText == "x1").Any());
+        }
+
+        [Fact]
+        public void ElementAccess_02()
+        {
+            var text = @"
+public class Cls
+{
+    public static void Main()
+    {
+        var x = new [] {1};
+        Test2(x[out int x1], x1);
+    }
+
+    static void Test2(object x, object y)
+    {
+        System.Console.WriteLine(y);
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(text,
+                                                            options: TestOptions.ReleaseExe,
+                                                            parseOptions: TestOptions.Regular.WithOutVarFeature());
+
+            compilation.VerifyDiagnostics(
+                // (7,21): error CS1525: Invalid expression term 'int'
+                //         Test2(x[out int x1], x1);
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "int").WithArguments("int").WithLocation(7, 21),
+                // (7,25): error CS1003: Syntax error, ',' expected
+                //         Test2(x[out int x1], x1);
+                Diagnostic(ErrorCode.ERR_SyntaxError, "x1").WithArguments(",", "").WithLocation(7, 25),
+                // (7,25): error CS0103: The name 'x1' does not exist in the current context
+                //         Test2(x[out int x1], x1);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "x1").WithArguments("x1").WithLocation(7, 25),
+                // (7,30): error CS0103: The name 'x1' does not exist in the current context
+                //         Test2(x[out int x1], x1);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "x1").WithArguments("x1").WithLocation(7, 30)
+                );
+
+            var tree = compilation.SyntaxTrees.Single();
+            Assert.False(tree.GetRoot().DescendantNodes().OfType<ArgumentSyntax>().Where(p => p.Identifier.ValueText == "x1").Any());
+        }
```

Comment:
Possible test ideas: 

- `int x; M(out var x);` // should fail because x is already defined
- M(out var var);